src/it/process-jar/pom.xml

```diff
@@ -0,0 +1,115 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.jenkins-ci.plugins</groupId>
+    <artifactId>plugin</artifactId>
+    <version>1.424</version>
+  </parent>
+
+  <groupId>org.jenkins-ci.tools.hpi.its</groupId>
+  <artifactId>process-jar</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <packaging>hpi</packaging>
+
+  <name>MyNewPlugin</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.6.3</version>
+      <optional>true</optional>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>@project.groupId@</groupId>
+          <artifactId>@project.artifactId@</artifactId>
+          <version>@project.version@</version>
+        </plugin>
+        <plugin>
+          <groupId>org.kohsuke</groupId>
+          <artifactId>access-modifier-checker</artifactId>
+          <version>1.0</version>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>@project.groupId@</groupId>
+        <artifactId>@project.artifactId@</artifactId>
+        <version>@project.version@</version>
+        <executions>
+          <execution>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>jar</goal>
+            </goals>
+            <!-- TODO the shade plugin doesn't like shading the non-main attached artifacts yet -->
+            <!--<configuration>-->
+              <!--<jarClassifier>unshaded</jarClassifier>-->
+            <!--</configuration>-->
+          </execution>
+        </executions>
+        <configuration>
+          <jarClassifier>shaded</jarClassifier>
+        </configuration>
+      </plugin>
+      <!-- TODO the shade plugin doesn't like shading the non-main attached artifacts yet -->
```

Comment:
Packaging is always after prepare-packaging but it's who sets first that can win 

Sent from my iPhone

> On 23 Oct 2015, at 5:49 p.m., Kanstantsin Shautsou <notifications@github.com> wrote:
> 
> In src/it/process-jar/pom.xml:
> 
> > +          <execution>
> > +            <phase>prepare-package</phase>
> > +            <goals>
> > +              <goal>jar</goal>
> > +            </goals>
> > +            <!-- TODO the shade plugin doesn't like shading the non-main attached artifacts yet -->
> > +            <!--<configuration>-->
> > +              <!--<jarClassifier>unshaded</jarClassifier>-->
> > +            <!--</configuration>-->
> > +          </execution>
> > +        </executions>
> > +        <configuration>
> > +          <jarClassifier>shaded</jarClassifier>
> > +        </configuration>
> > +      </plugin>
> > +      <!-- TODO the shade plugin doesn't like shading the non-main attached artifacts yet -->
> Can hpi somehow reorder packaging phase to be always latest?
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/maven-hpi-plugin/pull/24/r42887154%40github.com.